### PR TITLE
docs: update README with more contextual eval info

### LINKED ISSUE #129: Update README with deep-dive into Eval methods
Can draw from existing documentation elsewhere

### PR DESCRIPTION
**Pull request by nathan-weinberg**
Resolves #129

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -7,7 +7,74 @@
...
+
+You can read more about MMLU [here](https://arxiv.org/abs/2306.05685)
+
+### MMLU Branch
+
+MMLU Branch is an adaptation of MMLU that is designed to test custom knowledge that is being added to the model via a Git branch of the [taxonomy](https://github.com/instructlab/taxonomy).
+
+To conduct MMLU Branch, we create new tasks that are specific to the topic that is covered within the Pull Request. The teacher model `Mixtral-8x-7b-instruct` is used to generate new multiple choice questions based on the knowledge document included in a PR. A “task” is then constructed that references the newly generated questions and choices. These tasks are then used to judge the model’s grasp on new knowledge the same way MMLU works. Generation of these tasks are done as part of the [InstructLab SDG](https://github.com/instructlab/sdg) library.
+
+## MT-Bench / MT-Bench Branch Testing Steps
```
Comment (jaideepr97) [inline README.md:77]: is this supposed to be filled out?

Comment (nathan-weinberg) [reply README.md:77]: Oh yeah this needs some fixing up, thanks for pointing this out!

**Review by kelbrown20 — COMMENTED** [review]
Let me know your thoughts@ @nathan-weinberg

File: README.md [hunk 2]
```diff
@@ -7,7 +7,74 @@
...
+not get obsolete with time.
+
+You can read more about MT-Bench [here](https://arxiv.org/abs/2306.05685)
+
+### MT-Bench Branch
+
+MT-Bench Branch is an adaptation of MT-Bench that is designed to test custom skills that are added/being added to the model via the InstructLab project. These new skills
+come in the form of question/answer pairs in a Git branch of the [taxonomy](https://github.com/instructlab/taxonomy).
+
+MT-Bench Branch uses the user supplied seed questions to have the candidate model generate answers to, which are then judged by the judge model.
```
Comment (danmcp) [inline README.md:53]: Should we refer to using the answers from the qnas as providing the reference answers here?

File: README.md [hunk 3]
```diff
@@ -7,7 +7,74 @@
...
+
+InstructLab uses an implementation found [here](https://github.com/EleutherAI/lm-evaluation-harness) for running MMLU.
+
+You can read more about MMLU [here](https://arxiv.org/abs/2306.05685)
+
+### MMLU Branch
+
+MMLU Branch is an adaptation of MMLU that is designed to test custom knowledge that is being added to the model via a Git branch of the [taxonomy](https://github.com/instructlab/taxonomy).
+
+A teacher model is used to generate new multiple choice questions based on the knowledge document included in the taxonomy Git branch. A “task” is then constructed that references the newly generated questions and choices. These tasks are then used to judge the model’s grasp on new knowledge the same way MMLU works. Generation of these tasks are done as part of the [InstructLab SDG](https://github.com/instructlab/sdg) library.
```
[the remark this answers is not in the document]
Comment (danmcp) [reply README.md:75]: They are multiple choice questions.  So "generated questions and choices" is correct.  Maybe "answer choices" would be more correct?

Comment (danmcp) [inline README.md:75]: Would "score the model's grasp" be more descriptive than judge in this case?

**Review by danmcp — CHANGES_REQUESTED** [review]

File: README.md [hunk 4]
```diff
@@ -7,7 +7,74 @@
 
 Python Library for Evaluation
 
-## MT-Bench / MT-Bench-Branch Testing Steps
+## What is Evaluation?
+
+Evaluation is the step that allows us to gauge how a given model is performing against a set of specific tasks, by running known and standardized benchmark tests against
```
[the remark this answers is not in the document]
Comment (jaideepr97) [reply README.md:12]: we run multiple benchmarking tests @kelbrown20

File: README.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (kelbrown20) [reply README.md:75]: Yeah I think "answer choices" is a bit more clear. Thank you!

File: README.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (nathan-weinberg) [reply README.md:12]: Updated the text account for both these points

File: README.md [hunk 5]
```diff
@@ -7,7 +7,74 @@
...
+the model. Running this step gives us numerical scores across these various benchmarks, as well as logged excerpts/samples of the outputs the model produced during these
+benchmarks. Using a combination of these artifacts as reference, along with manual smoke screening of a model allows us to get the best idea about whether or not a model
+has learned and improved on something we are trying to teach it. There are 2 stages in the InstructLab process where we perform model evaluation:
+
+### Inter-checkpoint Evaluation
+
+This step occurs during multi-phase training. Each phase of training produces a number of different “checkpoints” of the model that are taken at various stages during
+the phase. At the end of each phase, we need to evaluate all the checkpoints in order to find the one that provides the best results. This is done as part of the
+[InstructLab Training](https://github.com/instructlab/training) library.
```
[the remark this answers is not in the document]
Comment (nathan-weinberg) [reply README.md:21]: I feel like that's something better put in the Training library README

File: README.md [hunk 6]
```diff
@@ -7,7 +7,74 @@
...
+**tl;dr** Full model evaluation of performance on **skills**
+
+MT-Bench is a type of benchmarking that involves asking a model 80 multi-turn questions - i.e.
+
+```text
+<Question 1> → <model’s answer 1> → <Follow-up question> → <model’s answer 2>
+```
+
+and having a “judge” model review the given multi-turn question, the provided model answer, and rate the answer with a score out of 10. The scores are then averaged out
+and the final score produced is the “MT-bench score” for that model. This benchmark assumes no factual knowledge on the model’s part. The questions are static, but do
```
[the remark this answers is not in the document]
Comment (nathan-weinberg) [reply README.md:43]: Basically just that the questions won't change over time - as for the "model" versus "checkpoint", evaluation using MT-Bench isn't _necessarily_ always run on a checkpoint, but it's always run on a model.

File: README.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (nathan-weinberg) [reply README.md:75]: Done!

Comment (nathan-weinberg) [reply README.md:75]: Yes I think so! Updating to that

File: README.md (hunk 2 above)
Comment (nathan-weinberg) [reply README.md:53]: Added a note on this, let me know what you think!

File: README.md [hunk 7]
```diff
@@ -7,7 +7,74 @@
...
+
+MT-Bench Branch uses the user supplied seed questions to have the candidate model generate answers to, which are then judged by the judge model using the user supplied
+seed answers as a reference.
+
+### Massive Multitask Language Understanding (MMLU)
+
+**tl;dr** Full model evaluation of performance on **knowledge**
+
+MMLU is a type of benchmarking that involves a series of fact-based multiple choice questions, along with 4 options for answers. It tests if a model is able to interpret
+the questions correctly, along the answers, formulate its own answer, then selects picks the correct option out of the provided ones. The questions are designed as a set
```
Comment (danmcp) [inline README.md:60]: nit: selects picks

**Review by danmcp — CHANGES_REQUESTED** [review]

File: README.md (hunk 7 above)
Comment (nathan-weinberg) [reply README.md:60]: good catch!

**Review by danmcp — APPROVED** [review]

File: README.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (kelbrown20) [reply README.md:21]: I'm good with that! I think I'll add types of training to my Training README updates

File: README.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (kelbrown20) [reply README.md]: Ah gotcha! Okay, yeah that makes sense, ty for clarifying

**Pull request merged**